docs/security/requirements.mdx

```diff
@@ -0,0 +1,120 @@
+---
+sidebar_position: 4
+---
+
+# Requirements
+
+The requirements in this section are organized hierarchically, with top-level requirements defining
+the core rules and obligations that must be met, serving as broad objectives for Bitwarden's
+security model. Sub-requirements expand on these by addressing specific scenarios, exceptions, or
+clarifications, and may override their parent requirement when explicitly stated. Sibling
+requirements at the same level must remain independent and free of contradictions.
+
+## Vault data (VD)
+
+:::warning Draft - Early version
+
+This section is still in its early stages and does not yet reflect current or future standards.
+
+:::
+
+1. Vault data **MUST** be protected _at rest_.
+
+   - a. The Client **MUST** encrypt vault data stored on disk.
+   - b. The Client **MUST** use the UserKey to encrypt vault data stored on the Server.
+   - c. The Client **SHOULD** ensure that no mechanisms exists such that vault data may be stored to
+     disk unencrypted without user consent. This includes cases where it is not the Client itself
+     that stores the data to disk.
+   - d. The Client **MUST** encrypt any vault data derivatives that can be used to re-create the
+     original form or are considered to be vault data on their own.
+   - e. The Client **MUST NOT** store any artifacts (e.g. encryption keys) on disk such that the
+     encrypted vault data can be decrypted without any additional information provided by the User.
+     - i. The Client **MAY** store such artifacts if given an _informed and explicit consent_ by the
+       User.
+   - f. The Client **MUST NOT** store any artifacts (e.g. encryption keys) such that the vault data
+     can be decrypted by the Server, regardless of consent.
+
+2. Vault data **MAY** be unprotected while _in use_.
+
+   - a. The Client **MAY** decrypt all vault data during vault unlock.
+     - i. The Client **SHOULD** minimize the quantity of decrypted vault data.
+   - b. The Client **MAY** leave vault data encrypted after unlock.
+   - c. The Client **MUST** ensure that unprotected data does not linger when no longer _in use_.
```

Comment:
Hmmm, I think `is not present in memory` is what I'm looking for here, but that makes this requirement hard to implement. Maybe we should change it to a `SHOULD` at the same time

```suggestion
   - c. The Client **SHOULD** ensure that unprotected data is not present in memory when no longer _in use_.
```

I still think this requirement will have a pretty large impact if adopted since it tends to discourage our huge caches, but having no caches is the current approach in the SDK so hopefully we won't have to prioritize any additional work just for this

Reply:
I agree with this; being more specific with the requirement while loosening the language a little (MUST -> SHOULD) seems like a good balance given that we currently decrypt everything at once in clients.